pkgs/development/python-modules/generic/default.nix

```diff
@@ -79,86 +68,47 @@ python.stdenv.mkDerivation (builtins.removeAttrs attrs ["disabled"] // {
     export DETERMINISTIC_BUILD=1
 
-    # prepend following line to import setuptools before distutils
-    # this way we make sure setuptools monkeypatches distutils commands
-    # this way setuptools provides extra helpers such as "python setup.py test"
-    sed -i '0,/import distutils/s//import setuptools;import distutils/' setup.py
-    sed -i '0,/from distutils/s//import setuptools;from distutils/' setup.py
-
     runHook postConfigure
   '';
 
-  checkPhase = attrs.checkPhase or ''
-      runHook preCheck
-
-      ${python}/bin/${python.executable} setup.py test
-
-      runHook postCheck
-  '';
-
   buildPhase = attrs.buildPhase or ''
     runHook preBuild
-
-    ${python}/bin/${python.executable} setup.py build ${lib.concatStringsSep " " setupPyBuildFlags}
-
+    cp ${setuppy} nix_run_setup.py
+    ${python.interpreter} nix_run_setup.py ${lib.optionalString (setupPyBuildFlags != []) ("build_ext " + (lib.concatStringsSep " " setupPyBuildFlags))} bdist_wheel
     runHook postBuild
   '';
 
   installPhase = attrs.installPhase or ''
     runHook preInstall
 
-    mkdir -p "$out/lib/${python.libPrefix}/site-packages"
-
-    export PYTHONPATH="$out/lib/${python.libPrefix}/site-packages:$PYTHONPATH"
-
-    ${python}/bin/${python.executable} setup.py install \
-      --install-lib=$out/lib/${python.libPrefix}/site-packages \
-      --old-and-unmanageable \
-      --prefix="$out" ${lib.concatStringsSep " " setupPyInstallFlags}
-
-    # --install-lib:
-    # sometimes packages specify where files should be installed outside the usual
-    # python lib prefix, we override that back so all infrastructure (setup hooks)
-    # work as expected
-
-    # --old-and-unmanagable:
-    # instruct setuptools not to use eggs but fallback to plan package install
-    # this also reduces one .pth file in the chain, but the main reason is to
-    # force install process to install only scripts for the package we are
-    # installing (otherwise it will install scripts also for dependencies)
-
-    # A pth file might have been generated to load the package from
-    # within its own site-packages, rename this package not to
-    # collide with others.
-    eapth="$out/lib/${python.libPrefix}"/site-packages/easy-install.pth
-    if [ -e "$eapth" ]; then
-        # move colliding easy_install.pth to specifically named one
-        mv "$eapth" $(dirname "$eapth")/${name}.pth
-    fi
+    mkdir -p "$out/${python.sitePackages}"
+    export PYTHONPATH="$out/${python.sitePackages}:$PYTHONPATH"
 
-    # Remove any site.py files generated by easy_install as these
-    # cause collisions. If pth files are to be processed a
-    # corresponding site.py needs to be included in the PYTHONPATH.
-    rm -f "$out/lib/${python.libPrefix}"/site-packages/site.py*
+    pushd dist
+    ${bootstrapped-pip}/bin/pip install *.whl --no-index --prefix=$out --no-cache
+    popd
 
     runHook postInstall
   '';
 
+  doInstallCheck = doCheck;
+  doCheck = false;
```

Comment:
As long as the set is not recursive ('rec' keyword) this works. But it is really calling for problems in the future and for readability.
Maybe put doCheck in a temporary variable or move these two lines in another merged by itself ?

```nix
{ /*...*/ } // {
  doInstallCheck = doCheck;
} // {
  doCheck = false;
  #...
}
```